cookiecutter/vcs.py

```diff
@@ -15,34 +18,263 @@
 logger = logging.getLogger(__name__)
 
 
-BRANCH_ERRORS = [
-    'error: pathspec',
-    'unknown revision',
+class VCS:
+    """Abstract base VCS class. Handles the cloning of VCS repositories."""
+
+    cmd = 'xyz'
```

Comment:
It seems more obvious to make this `None` and then raise an `AttributeError` in `__init__` or `__new__` if it is not set.